src/models.rs

```diff
@@ -568,3 +639,485 @@ pub struct Generic {
     pub spec: serde_json::Value,
     pub status: Option<serde_json::Value>,
 }
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum NumberOrString {
+    Number(i64),
+    String(String),
+}
+
+impl NumberOrString {
+    pub fn as_number(&self) -> Option<i64> {
+        match self {
+            NumberOrString::Number(n) => Some(*n),
+            NumberOrString::String(s) => {
+                // Parse numeric part and unit suffix
+                let numeric: String = s.chars().take_while(|c| c.is_digit(10)).collect();
+                let unit = s[numeric.len()..].to_lowercase().replace(" ", "");
+                let base: Option<i64> = numeric.parse().ok();
+                base.map(|n| match unit.as_str() {
+                    "kb" | "k" => n * 1024,
+                    "mb" | "m" => n * 1024 * 1024,
+                    "gb" | "g" => n * 1024 * 1024 * 1024,
+                    "tb" | "t" => n * 1024 * 1024 * 1024 * 1024,
+                    "min" => n * 60,
+                    "hr" | "h" => n * 60 * 60,
+                    "day" | "d" => n * 60 * 60 * 24,
+                    "cpu" | "cpus" | "core" | "cores" => n * 1000,
+                    "gpu" | "gpus" => n * 1000,
+                    "mcpu" | "mcpus" | "millicpu" | "millicpus" => n,
+                    "mgpu" | "mgpus" | "milligpu" | "milligpus" => n,
+                    "mcore" | "mcores" | "millicore" | "millicores" => n,
+                    _ => n,
+                })
+            }
+        }
+    }
+
+    pub fn as_string(&self) -> String {
+        match self {
+            NumberOrString::Number(n) => n.to_string(),
+            NumberOrString::String(s) => s.clone(),
+        }
+    }
+}
+
+impl From<i64> for NumberOrString {
+    fn from(val: i64) -> Self {
+        NumberOrString::Number(val)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json::json;
+
+    #[test]
+    fn test_number_or_string() {
+        // Test raw number
+        let num = serde_json::from_value::<NumberOrString>(json!(1234)).unwrap();
+        assert_eq!(num.as_number(), Some(1234));
+
+        // Test raw number
+        let num = serde_json::from_value::<NumberOrString>(json!("1234")).unwrap();
+        assert_eq!(num.as_number(), Some(1234));
+
+        // Test string with kb unit
+        let str = serde_json::from_value::<NumberOrString>(json!("1234kb")).unwrap();
+        assert_eq!(str.as_number(), Some(1234 * 1024));
+        assert_eq!(str.as_string(), "1234kb");
+
+        // Test string with mb unit
+        let str = serde_json::from_value::<NumberOrString>(json!("1234mb")).unwrap();
+        assert_eq!(str.as_number(), Some(1234 * 1024 * 1024));
+        assert_eq!(str.as_string(), "1234mb");
+
+        // Test string with min unit
+        let str = serde_json::from_value::<NumberOrString>(json!("60min")).unwrap();
+        assert_eq!(str.as_number(), Some(60 * 60));
+        assert_eq!(str.as_string(), "60min");
```

Comment:
Especially with time (but possibly with storage space as well), I'd maybe test also combinations like "1h 47min 3 sec". Would "3kb 3M" parse ? What about "7min 8gpus" :see_no_evil:

Reply:
"7min 8gpus" is probably something for the future but for time we can use humantime crate which does exactly what you describe

Reply:
Would it then make sense to split different types (CPUs/GPUs, MEM, Time) and possibly have an enum including them, with some brute force parsing logic trying each one (or w/ regex match based heuristics)?

Reply:
When removing the time cases (which I did), the rest of the units isnt that much, so I think its fine like this. We can always extend it, but I feel it does its job well.